client/asset/zec/shielded_rpc.go

```diff
@@ -52,7 +67,153 @@ type unifiedReceivers struct {
+}
+
+// z_validateaddress "address"
+func zValidateAddress(c rpcCaller, addr string) (res *zValidateAddressResult, err error) {
+	return res, c.CallRPC(methodZValidateAddress, []interface{}{addr}, &res)
+}
+
+type zSendManyRecipient struct {
+	Address string  `json:"address"`
+	Amount  float64 `json:"amount"`
+	Memo    string  `json:"memo,omitempty"`
+}
+
+func singleSendManyRecipient(addr string, amt uint64) []*zSendManyRecipient {
+	return []*zSendManyRecipient{
+		{
+			Address: addr,
+			Amount:  btcutil.Amount(amt).ToBTC(),
+		},
+	}
+}
+
+// When sending using z_sendmany and you're not sending a shielded tx to the
+// same pool, you must specify a lower privacy policy.
+type privacyPolicy string
+
+const (
+	FullPrivacy             privacyPolicy = "FullPrivacy"
+	AllowRevealedRecipients privacyPolicy = "AllowRevealedRecipients"
+	AllowRevealedAmounts    privacyPolicy = "AllowRevealedAmounts"
+	AllowRevealedSenders    privacyPolicy = "AllowRevealedSenders"
+)
+
+// z_sendmany "fromaddress" [{"address":... ,"amount":...},...] ( minconf ) ( fee ) ( privacyPolicy )
+func zSendMany(c rpcCaller, fromAddress string, recips []*zSendManyRecipient, priv privacyPolicy) (operationID string, err error) {
+	const minConf, fee = 1, 0.00001
+	return operationID, c.CallRPC(methodZSendMany, []interface{}{fromAddress, recips, minConf, fee, priv}, &operationID)
+}
+
+type operationStatus struct {
+	ID           string    `json:"id"`
+	Status       string    `json:"status"` // "success", "failed", others?
+	CreationTime int64     `json:"creation_time"`
+	Error        *struct { // nil for success
+		Code    int32  `json:"code"`
+		Message string `json:"message"`
+	} `json:"error" `
+	Result *struct {
+		TxID string `json:"txid"`
+	} `json:"result"`
+	ExecutionSeconds float64 `json:"execution_secs"`
+	Method           string  `json:"method"`
+	Params           struct {
+		FromAddress string `json:"fromaddress"`
+		Amounts     []struct {
+			Address string  `json:"address"`
+			Amount  float64 `json:"amount"`
+		} `json:"amounts"`
+		MinConf uint32  `json:"minconf"`
+		Fee     float64 `json:"fee"`
+	} `json:"params"`
+}
+
+// awaitSendManyOperation waits for the asynchronous result from a z_sendmany
+// operation.
+func awaitSendManyOperation(ctx context.Context, c rpcCaller, operationID string) (*chainhash.Hash, error) {
+	for {
+		res, err := zGetOperationResult(c, operationID)
+		if err != nil && !errors.Is(err, ErrEmptyOpResults) {
+			return nil, fmt.Errorf("error getting operation result: %w", err)
+		}
+		if res != nil {
+			if res.Status == "failed" {
+				return nil, fmt.Errorf("z_sendmany operation failed: %s", res.Error.Message)
+			}
+			if res.Status == "success" {
+				txid, err := chainhash.NewHashFromStr(res.Result.TxID)
+				if err != nil {
+					return nil, fmt.Errorf("error decoding txid: %w", err)
+				}
+				return txid, nil
+			}
+			fmt.Println("--###########$$$$$$$$$$$$$$$$$$ operation status", res.Status)
```

Comment:
`Debugf` or remove?  Have you seen this print though?  If it never seems to happen, maybe `Warnf`.